feat: Update server resource and add BYOC architecture doc (#58)
tensorchord/modelz-docs: pages/collaboration/team.mdx

# Team Collaboration

> After having signed up to ModelZ, you’ve automatically received a personal account that you can use to run and deploy personal models. To deploy collaborate with others, however, you can create or join additional teams.

Reviewer: ```suggestion
After registering with ModelZ, you will automatically have a personal account that can run and deploy personal models. To collaborate with others, you can create or join a team.
```

## Create a Team

> You can create a team, and fill in your team member's e-mail address and role. 

Reviewer: ```suggestion
You can create a team, invite team members using their email addresses, and assign roles to them.
```

> - If the email of team member is already signed up in ModelZ, the member will receive a team invitation by e-mail. After accepting the invitation, he can see the team in his team list.

Reviewer 2: Maybe `the email of team member` to be `the invited user`?

Author: lgtm

> - If email is not signed up in ModelZ, the member will receive a sign up email with invitation.

Reviewer 2: Same here.

Author: fixed.

![create_team](./images/create_team.jpg)

## Invite Team member

> To invite a new team member, first switch your user setting to team.

Reviewer 2: Do we need a guild and image of switch this setting?

![invite_team_member](./images/invite_team_member.jpg)

## Team ID and APIKey

> Each team has its own team id and apikey, which a team member can obtain in the following way, using SDK or CLI for deployment control and endpoint access.

Reviewer: ```suggestion
Each team has its own team ID and API key, which can be used by SDK or CLI for deployment control and endpoint access. They can be found in the following position.
```

![team_id_apikey](./images/team_id_apikey.jpg)

## Role
Teams have two different role with different levels of access privileges:
- Owner
- Member

> The user that creates a team is automatically set as the Owner for that team. The owner can assign any other roles within the team, as well as pending other members of the workspace.

Reviewer: ```suggestion
The user that creates the team is automatically set as the `Owner` of the team. The owner can assign any other roles within the team, as well as pending other members of the workspace.
```

What does "pending" here mean?

Author: Pending means the invited user not accept the invitation or the owner doesn't want the member to work on the team's resources anymore.

Reviewer: But "pending" is not a verb, nor a -ing here.

> The user that creates a team is automatically set as the Owner for that team. The owner can assign any other roles within the team, as well as pending other members of the workspace.

Reviewer 2: `pending other members`  or `other pending members`?

Author: thx, fixed.

> Member of a team can not assign any access privileges within the workspace but can otherwise perform any action like running and deploying models and modify Secrets.

Reviewer: ```suggestion
Member of a team can not assign any access privileges within the workspace but can perform actions like creating deployments and modifying Secrets.
```
